staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1beta1/types.go

```diff
@@ -57,3 +60,22 @@ type ExecCredentialStatus struct {
 	// PEM-encoded private key for the above certificate.
 	ClientKeyData string `json:"clientKeyData,omitempty"`
 }
+
+// Cluster contains information to allow an exec plugin to communicate with the kubernetes cluster being authenticated to.
+type Cluster struct {
+	// Server is the address of the kubernetes cluster (https://hostname:port).
+	Server string `json:"server"`
+	// ServerName is passed to the server for SNI and is used in the client to check server
+	// certificates against. If ServerName is empty, the hostname used to contact the
+	// server is used.
+	// +optional
+	ServerName string `json:"serverName,omitempty"`
+	// CAData contains PEM-encoded certificate authority certificates.
+	// If empty, system roots should be used.
+	// +listType=atomic
+	// +optional
+	CAData []byte `json:"caData,omitempty"`
+	// Config holds additional config data that is specific to the exec plugin with regards to the cluster being authenticated to.
```

Comment:
From this documentation, it's not obvious how or why I should use this. Where does this data come from? Can you add details on how this should be used? Should it contain secrets? Can you add examples of reasonable usage like "audience"?

Reply:
Added some details.